src/KubernetesClient/KubernetesClientConfiguration.ConfigFile.cs

```diff
@@ -557,13 +558,38 @@ public static ExecCredentialResponse ExecuteExternalCommand(ExternalExecution co
                 throw new KubeConfigException($"external exec failed due to: {ex.Message}");
             }
 
-            var stdout = process.StandardOutput.ReadToEnd();
-            var stderr = process.StandardError.ReadToEnd();
-            if (string.IsNullOrWhiteSpace(stderr) == false)
+            var sb = new StringBuilder();
+            var buffer = new char[1024];
+            while (true)
+            {
+                var readTask = process.StandardError.ReadAsync(buffer, 0, buffer.Length);
+
+                if (readTask.Wait(TimeSpan.FromSeconds(2)))
```

Comment:
exec will still hang if the process is consistently sending output, right?

Reply:
That is correct, if the external application keeps sending errors. This will continue looping until it stops for two seconds.

My only experience has been with Kubelogin. I'm not sure how other clients behave here.

Reply:
link #1092

Reply:
@tg123, thanks for linking that issue. I used the Event based option from that Issue and updated the PR.

Now, the process will timeout after 5 seconds no matter what, then we check if stderror has a value, if it does we will throw the response as an exception otherwise we throw a timeout exception.

The only gap remaining is the process will stick around on Errors indefinitely. We might need to add some more code to store the processes and kill them after 5 minutes or something. Let me know if you want me to code this in.